bin/node/testing/src/bench.rs

```diff
@@ -56,6 +56,7 @@ use sc_client_api::{
 	execution_extensions::{ExecutionExtensions, ExecutionStrategies},
 };
 use sp_core::{Pair, Public, sr25519};
+use sc_block_builder::BlockBuilderProvider;
```

Comment:
Ideally `traits` should be defined in `sp_` crates, to avoid bringing the entire implementation together with the trait. Is it possible to move `trait BlockBuilderProvider` to `sp-block-builder` without violating dependency rules?

Reply:
Then we'd have to move the `BlockBuilder` struct there as well because it's used in the trait and seeing as crates in `primtives/*` can't depend on crates in `client/*`, I would just elect for us to merge `sc-block-builder` and `sp-block-builder`.

Reply:
Ok after messing around with this, I've discovered I can't move the trait to `sp_block_builder`.

moving it would require moving the `BlockBuilder` struct which would also require moving the `BuiltBlock` struct which is generic over a `StateBackend` from `sc_client_api`.

and `primitives` -> `client` dependencies are discouraged.